core/modules/user/user.entity.inc

```diff
@@ -495,4 +495,61 @@ class UserStorageController extends EntityDatabaseStorageController {
       }
     }
   }
+
+  /**
+   * Implements EntityControllerInterface::buildContent().
+   */
+  public function buildContent(EntityInterface $account, $view_mode = 'full', $langcode = NULL, $context = array()) {
```

Comment:
$account is an instance of User. The User class extends Entity (which implements EntityInterface), but the User class also implements UserInterface, which is more specialized than EntityInterface. 

I'm thinking it may be best to declare the argument as UserInterface to be clearer.

Reply:
Actually I'm wrong here. The argument declaration has to match the method it's extending